instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/log4j/contextdata/v2_17/OpenTelemetryContextDataProvider.java

```diff
@@ -77,9 +77,15 @@ public Map<String, String> supplyContextData() {
     contextData.putAll(staticContextData);
 
     SpanContext spanContext = currentSpan.getSpanContext();
-    contextData.put(TRACE_ID, spanContext.getTraceId());
-    contextData.put(SPAN_ID, spanContext.getSpanId());
-    contextData.put(TRACE_FLAGS, spanContext.getTraceFlags().asHex());
+    contextData.put(
+        TRACE_ID_KEY == null ? LoggingContextConstants.TRACE_ID : TRACE_ID_KEY,
+        spanContext.getTraceId());
+    contextData.put(
+        SPAN_ID_KEY == null ? LoggingContextConstants.SPAN_ID : SPAN_ID_KEY,
+        spanContext.getSpanId());
+    contextData.put(
+        TRACE_FLAGS_KEY == null ? LoggingContextConstants.TRACE_FLAGS : TRACE_FLAGS_KEY,
+        spanContext.getTraceFlags().asHex());
```

Comment:
I think it would be better to add a version of `ConfigPropertiesUtil.getString` that returns default value when value is not present. Similar methods already exists for boolean and int. Then you could use 
```java 
  private static final String TRACE_ID_KEY =
      ConfigPropertiesUtil.getString("otel.instrumentation.common.logging.trace-id", LoggingContextConstants.TRACE_ID);
```

Reply:
> I think it would be better to add a version of `ConfigPropertiesUtil.getString` that returns default value when value is not present. Similar methods already exists for boolean and int. Then you could use
> 
> ```java
>   private static final String TRACE_ID_KEY =
>       ConfigPropertiesUtil.getString("otel.instrumentation.common.logging.trace-id", LoggingContextConstants.TRACE_ID);
> ```

I had the same thought when writing this piece of code.